backend/lib/models/Notification.js

```diff
@@ -0,0 +1,50 @@
+const { Schema, model } = require("mongoose");
+
+const notificationSchema = new Schema(
+  {
+    action: {
+      enum: ["like", "comment", "post"],
+      required: true,
+      type: String,
+    },
+    post: {
+      id: String,
+      title: String,
+    },
+    receiver: {
+      ref: "User",
+      required: true,
+      type: Schema.Types.ObjectId,
+    },
+    seenAt: Date,
+    sentAt: Date,
+    triggeredBy: {
+      id: String,
```

Comment:
```suggestion  id: {
    ref: "User",
    type: ObjectId,
  },
```

This should be a `ObjectId` `ref`

Reply:
Maybe we can avoid the additional queries required to return the referenced documents, since notifications have a 15 days TTL and are sent shortly, we shouldn't worry much about changes in post or triggeredBy, but I could be wrong.

Reply:
Are referenced documents always queried (I guess this is eager loading)? I am not familiar with Mongo or Mongoose. Is there a concept of "lazy loading" where the referenced documents are only fetched if the field is accessed?

Reply:
Ah, found the relevant Mongoose doc about eager loading: https://mongoosejs.com/docs/populate.html
By default, the related documents are not fetched unless you call `populate()`. But it is still an extra query if we use it. I guess there is no way around the N+1 query problem due to the nature of MongoDB.

Should we just include the ref as well as the other three fields? Like @Zaydme said, I don't think it matters too much that the name/photo changes for notifications.

Reply:
A bit late here but:

You're right it will only query the referenced doc if you call `populate()` on that field.
Populate should make a single query for all referenced docs, the N+1 problem is just if you want to call `populate()` for an array of docs or something similar. Then if that is something usual in this use case you can just create a custom method in the model that does a single custom lookup and get all the data with a single query.